lstchain/tools/lstchain_create_drs4_pedestal_file.py

```diff
@@ -0,0 +1,109 @@
+"""
+Create drs4 time pedestal fits file.
+"""
+import numpy as np
+from astropy.io import fits
+
+from ctapipe.core import Provenance, traits
+from ctapipe.core import Tool
+from ctapipe.io import EventSource
+from lstchain.calib.camera.r0 import LSTR0Corrections
+from lstchain.calib.camera.drs4 import DragonPedestal
+
+
+class PedestalFITSWriter(Tool):
+
+    name = "PedestalFITSWriter"
+    description = "Generate a pedestal FITS file"
+
+    output_file = traits.Path(
+        default_value="pedestal.fits",
+        help="Path to the generated FITS pedestal file",
+        directory_ok=False,
+        exists=False,
+    ).tag(config=True)
+
+    deltaT = traits.Bool(
+        help="Flag to use deltaT correction. Default=True", default_value=True
+    ).tag(config=True)
+
+    aliases = {
+        "input_file": "EventSource.input_url",
+        "output_file": "PedestalFITSWriter.output_file",
+    }
+
+    flags = {
+        "deltaT": ({"PedestalFITSWriter": {"deltaT": True}}, "Flag to use deltaT correction. Default is True")
+    }
+
+    classes = [EventSource, LSTR0Corrections, DragonPedestal]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a pedestal FITS file for low level calibration.
+
+        For getting help run:
+        lstchain_create_drs4_pedestal_file --help
+        """
+
+        self.eventsource = None
+        self.pixel_ids = None
+        self.pedestal = None
+        self.lst_r0 = None
+
+    def setup(self):
+
+        self.log.debug("Opening file")
+        self.eventsource = EventSource.from_config(parent=self)
+        self.lst_r0 = self.add_component(LSTR0Corrections(parent=self))
+
+        for event in self.eventsource:
+            tel_id = event.r0.tels_with_data[0]
+            self.pixel_ids = event.lst.tel[tel_id].svc.pixel_ids
+            self.pedestal = DragonPedestal(tel_id=tel_id, n_module=event.lst.tel[tel_id].svc.num_modules)
+            break
+
+    def start(self):
+
+        try:
+            if self.deltaT:
+                self.log.info("DeltaT correction active")
+                for i, event in enumerate(self.eventsource):
+                    for tel_id in event.r0.tels_with_data:
+                        self.lst_r0.time_lapse_corr(event, tel_id)
+                        self.pedestal.fill_pedestal_event(event)
+                        if i % 500 == 0:
+                            self.log.debug(f"i = {i}, ev id = {event.index.event_id}")
+            else:
+                self.log.info("DeltaT correction no active")
+                for i, event in enumerate(self.eventsource):
+                    self.pedestal.fill_pedestal_event(event)
+                    if i % 500 == 0:
```

Comment:
Just use `tqdm` around the for loop, it will print a nice progress bar. See e.g. the stage 1 tool here:
https://github.com/cta-observatory/ctapipe/blob/a5ad0f181a2f1565324ce6af5c18a806c9b33dd5/ctapipe/tools/stage1.py#L125